kbase-extension/static/kbase/js/common/cellComponents/filePathWidget.js

```diff
@@ -313,129 +334,136 @@ define([
-                                        .then((widget) => {
-                                            widgets.push(widget);
-                                            return widget.start({
-                                                node: document.getElementById(filePathParams.view[spec.id].id)
-                                            });
-                                        });
-                                } catch (ex) {
-                                    console.error('Error making input field widget', ex);
-                                    const errorDisplay = div({
-                                        class: 'kb-field-widget__error_message--file-paths'
-                                    }, [
-                                        ex.message
-                                    ]);
-                                    document.getElementById(filePathParams.view[spec.id].id).innerHTML = errorDisplay;
-                                }
-                            }));
-                        }
-                    }).then(function() {
-                    //     TODO: commenting out for now as we need to figure out styling for the row number and trash icon. Should have a convo with design about how they want to handle given the current designs only show how to handle one or two file inputs, and in our current example we have 3 file inputs + 1 output
-                    //     $(places.parameterFields).prepend(
-                    //         span({
-                    //             class: `${cssBaseClass}__file_number`,
-                    //         }, [
-                    //             '1'
-                    //         ])
-                    //     );
-                    //     $(places.parameterFields).append(
-                    //         span({
-                    //             class: `${cssBaseClass}__icon_cell--trash`,
-                    //         }, [
-                    //             span({
-                    //                 class: `${cssBaseClass}__icon--trash fa fa-trash-o fa-lg`,
-                    //             })
-                    //         ])
-                    //     );
-                    });
+            }).map(function (id) {
+                return params.specs[id];
             });
         }
 
+        function createParamWidget(appSpec, filePathParams, parameterId) {
+            const spec = filePathParams.paramMap[parameterId];
+            try {
+                return makeFieldWidget(appSpec, spec, initialParams[spec.id], filePathParams.layout)
+                    .then((widget) => {
+                        widgets.push(widget);
+                        return widget.start({
+                            node: document.getElementById(filePathParams.view[spec.id].id)
+                        });
+                    });
+            } catch (ex) {
```

Comment:
Can we update this so makeFieldWidget has proper Promise error handling, and then get rid of this try/catch block entirely?